packages/flutter_tools/lib/executable.dart

```diff
@@ -179,8 +179,11 @@ Future<Null> _exit(int code) async {
   logger.flush();
 
   // Give the task / timer queue one cycle through before we hard exit.
-  await Timer.run(() {
+  Completer<Null> completer = new Completer<Null>();
+  Timer.run(() {
```

Comment:
> Using a Timer will mean that futures with completion handlers that themselves await things will have time to finish too.

I think that `new Future(() => ...)` is implemented in terms of `Timer.run`.

> Sounds like what you're saying is that I should just remove the await, complete the _exit early (rather than never, as now), and add a comment explaining all this.

Sounds good; re-reading it now, I agree the await doesn't do anything.